pretyping/recordops.ml

```diff
@@ -202,7 +202,14 @@ let cs_pattern_of_constr env t =
       App (f,vargs) ->
 	begin
 	  try Const_cs (global_of_constr f) , None, Array.to_list vargs
-          with e when CErrors.noncritical e -> raise Not_found
+          with
+          | Not_found when isProj f ->
+              let p, c = destProj f in
+              let { Environ.uj_type = ty } = Typeops.infer env c in
```

Comment:
Why not retyping here?

Reply:
I don't know I've just adapted the code for `Proj` from below

Reply:
Oh I see.
Maybe it would be more uniform to make the function recursive, then the App case would just add the arguments to the list returned by a recursive call on the head. This would change behaviour on applied Rel (from Not_found to returning something), I don't know what the consequence of that would be.

Reply:
It looks to me like Retyping would be enough, t is supposed to be typable here I believe.

Reply:
I think this is because this is supposed to work on Constr when retyping works on EConstr.